engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/keys/ScatterKey.scala

```diff
@@ -59,14 +59,10 @@ private [execution] case class ScatterKey(node: ScatterNode) extends JobKey {
       * (not the PortBasedGraphOutputNode that is just a proxy, but the source node). If that node is a CallNode
       * we create only one collector for all those scatter gather ports. 
      */
-    val collectors = node.outputMapping.groupBy(_.outputToGather.source.graphNode) flatMap {
-      case (taskCallNode: CallNode, scatterGatherPorts) => List(ScatterCollectorKey(taskCallNode, scatterGatherPorts, node, count))
-        // For other types of nodes we don't assume any time locality regarding the output ports availability and create
-        // one collector per port
-      case (declarationNode: AnonymousExpressionNode, scatterGatherPorts) => scatterGatherPorts map { sgp => ScatterCollectorKey(declarationNode, Set(sgp), node, count) }
-      case (_: ConditionalNode, scatterGatherPorts) => scatterGatherPorts map { sgp => ScatterCollectorKey(sgp.outputToGather, Set(sgp), node, count) }
-      case _ => List.empty
-    }
+    val collectors: Set[ScatterCollectorKey] = (node.outputMapping.groupBy(_.outputToGather.source.graphNode) flatMap {
+      case (_: CallNode | _: ExposedExpressionNode | _: ConditionalNode, scatterGatherPorts) => scatterGatherPorts.map(sgp => ScatterCollectorKey(sgp, count))
```

Comment:
Why has the handling of anonymous vs exposed expression nodes changed?

Reply:
This is just fixing a typo. Previously anonymous expressions (eg call input evaluations) were being collected and exposed outside the scatter and declarations were not.

Reply:
oops